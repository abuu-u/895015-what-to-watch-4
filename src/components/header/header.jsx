import React from "react";
import PropTypes from "prop-types";
import {Link} from "react-router-dom";
import {AppRoute} from '../../const';
import {connect} from "react-redux";
import {getAuthInfo} from "../../reducer/user/selectors.js";

const Header = (props) => {
  const {authInfo, children} = props;

  return (
    <header className="page-header">
      <div className="logo">
        <Link
          to={AppRoute.ROOT}
          className="logo__link"
        >
          <span className="logo__letter logo__letter--1">W</span>
          <span className="logo__letter logo__letter--2">T</span>
          <span className="logo__letter logo__letter--3">W</span>
        </Link>
      </div>

      {children}

      <div className="user-block">
        {authInfo.avatarUrl
          ? <div className="user-block__avatar">
            <img src={`https://4.react.pages.academy${authInfo.avatarUrl}`} alt="User avatar" width="63" height="63" />
          </div>
          : <Link
            to={AppRoute.LOGIN}
            className="user-block__link"
          >Sign in</Link>}
      </div>
    </header>
  );
};

Header.propTypes = {
  authInfo: PropTypes.shape({
    avatarUrl: PropTypes.string.isRequired,
  }),
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]),
};

const mapStateToProps = (state) => ({
  authInfo: getAuthInfo(state),
});

export {Header};
export default connect(mapStateToProps)(Header);
