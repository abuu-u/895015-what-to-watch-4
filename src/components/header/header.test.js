import React from "react";
import renderer from "react-test-renderer";
import {Header} from "./header.jsx";

const authInfo = {
  avatarUrl: `img/1.png`
};

it(`Render unauthorized Header`, () => {
  const tree = renderer
    .create(<Header
      authorizationStatus={`NO_AUTH`}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`Render authorized Header`, () => {
  const tree = renderer
    .create(<Header
      authorizationStatus={`AUTH`}
      authInfo={authInfo}
    />, {
      createNodeMock() {
        return {};
      }
    })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
