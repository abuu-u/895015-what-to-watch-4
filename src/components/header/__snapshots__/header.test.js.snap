// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render authorized Header 1`] = `
<header
  className="page-header"
>
  <div
    className="logo"
  >
    <a
      className="logo__link"
      href="/"
      onClick={[Function]}
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
  </div>
  <div
    className="user-block"
  >
    <div
      className="user-block__avatar"
    >
      <a
        href="/mylist"
        onClick={[Function]}
      >
        <img
          alt="User avatar"
          height="63"
          src="https://4.react.pages.academyimg/1.png"
          width="63"
        />
      </a>
    </div>
  </div>
</header>
`;

exports[`Render unauthorized Header 1`] = `
<header
  className="page-header"
>
  <div
    className="logo"
  >
    <a
      className="logo__link"
      href="/"
      onClick={[Function]}
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
  </div>
  <div
    className="user-block"
  >
    <div
      className="user-block__avatar"
    >
      <a
        href="/mylist"
        onClick={[Function]}
      >
        <img
          alt="User avatar"
          height="63"
          src="https://4.react.pages.academyimg/1.png"
          width="63"
        />
      </a>
    </div>
  </div>
</header>
`;
